### ISSUE TITLE
implementation of creating a workspace on the frontend

### ISSUE THREAD

**Issue by D3PSI** [D3PSI/lumium#171]
the renderer wasm module exports the `async create_workspace(password)` function that handles the following tasks:
- it generates a cryptographically secure random workspace master key
- it generates cryptographically secure recovery codes
- it encrypts the workspace master key with the SHA256 of the provided password (the workspace password)
- it encrypts the workspace master key with the SHA256 of each recovery code
- it calls the `PUT /v1/secure/workspace` endpoint with all of the generated key variants
- it downloads the recovery codes as a text file to the client system

summary: when clicking the `Create workspace and download recovery codes` button we simply need to call this function _and `await` it_.

**Comment by D3PSI** [comment]
@ReddixT it might be worth noting that i have not in fact tested this. there is a chance that the wasm module is still buggy and calls a wrong endpoint. if that happens, please ping me on here

**Comment by ReddixT** [comment]
> @ReddixT it might be worth noting that i have not in fact tested this fully. there is a chance that the wasm module is still buggy and calls a wrong endpoint. if that happens, please ping me on here

![image](https://user-images.githubusercontent.com/63563893/207316964-1d7104b9-ddc2-42ad-84e8-e50c51f6494e.png)
the password shouldnt be a number

**Comment by D3PSI** [comment]
did you build latest develop?

> @ReddixT it might be worth noting that i have not in fact tested this fully. there is a chance that the wasm module is still buggy and calls a wrong endpoint. if that happens, please ping me on here

**Comment by ReddixT** [comment]
ye
![image](https://user-images.githubusercontent.com/63563893/207317632-70b527ed-cf53-4f80-8924-3702b8d08a55.png)


> did you build latest develop?
> 
> > @ReddixT it might be worth noting that i have not in fact tested this fully. there is a chance that the wasm module is still buggy and calls a wrong endpoint. if that happens, please ping me on here

yes
![image](https://user-images.githubusercontent.com/63563893/207317670-9c4fe762-6cc4-4def-89fc-6284c3d662ec.png)

**Comment by D3PSI** [comment]
interesting, that is... let me check the bindings that it generates

**Comment by ReddixT** [comment]
also running into this issue
![image](https://user-images.githubusercontent.com/63563893/207318841-40308568-d675-4eaa-ae30-1cfa8a37ce7d.png)

**Comment by D3PSI** [comment]
fixes have been pushed directly to develop for both issues

**Comment by ReddixT** [comment]
> fixes have been pushed directly to develop for both issues

![image](https://user-images.githubusercontent.com/63563893/207326536-d73de3cb-4910-4e35-bd62-2d94c28c39bd.png)
are you sure about that?

**Comment by D3PSI** [comment]
yes, in fact i am sure, because i tested it locally. the renderer only gets rebuilt when you run `yarn && yarn build` - which you should always do anyway :). only then will the newly built wasm binary be bundled in the frontend
